realm/realm-library/src/main/java/io/realm/internal/ManagedObject.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2017 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.realm.internal;
```

Comment:
I debated that.  In the end, I decided that unless there is a reason to make this public, we should make it private.  It does not seem to me to be an abstraction that is useful to clients.  If you have a reason to suggest otherwise, I could easily be swayed.